tools/serverpod_cli/bin/generator/class_generator_dart.dart

```diff
@@ -837,6 +843,138 @@ class ClassGeneratorDart extends ClassGenerator {
+                          if (column
+                              .shouldSerializeFieldForDatabase(serverCode))
+                            refer('ColumnDefinition',
+                                    serverpodProtocolUrl(serverCode))
+                                .call([], {
+                              'name': literalString(column.name),
+                              'columnType': refer(
+                                  'ColumnType.${column.type.databaseTypeEnum}',
+                                  serverpodProtocolUrl(serverCode)),
+                              // The id column is not null, since it is auto incrementing.
+                              'isNullable': literalBool(
+                                  column.name != 'id' && column.type.nullable),
+                              'dartType': literalString(column.type.toString()),
+                              if (column.name == 'id')
+                                'columnDefault': literalString(
+                                    "nextval('${classDefinition.tableName!}_id_seq'::regclass)"),
+                            }),
+                      ]),
+                      'foreignKeys': literalList([
+                        for (var i = 0;
+                            i <
+                                classDefinition.fields
+                                    .where((field) => field.parentTable != null)
+                                    .length;
+                            i++)
+                          () {
+                            var column = classDefinition.fields
+                                .where((field) => field.parentTable != null)
+                                .toList()[i];
+                            return refer('ForeignKeyDefinition',
+                                    serverpodProtocolUrl(serverCode))
+                                .call([], {
+                              'constraintName': literalString(
+                                  // Converting to lower case, since Serverpod does not quote constraint names in the generated SQL.
+                                  '${classDefinition.tableName!}_fk_$i'
+                                      .toLowerCase()),
+                              'columns': literalList([
+                                literalString(column.name),
+                              ]),
+                              'referenceTable':
+                                  literalString(column.parentTable!),
+                              'referenceColumns': literalList([
+                                literalString('id'),
+                              ]),
+                              'onUpdate': literalNull,
+                              'onDelete': refer('ForeignKeyAction.cascade',
+                                  serverpodProtocolUrl(serverCode)),
+                              'matchType': literalNull,
+                            });
+                          }(),
```

Comment:
This is a mapping with an index, alternatively, we could use [this method](https://pub.dev/documentation/collection/latest/collection/IterableExtension/mapIndexed.html) (additional dependency) or use something like `.toList().asMap().entries.map()`.